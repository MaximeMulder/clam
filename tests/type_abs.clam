type Tuple = [T] (Int, T);
type Make = [T] -> (T) -> (Int, T);

def make_zero: Make = [T] => (p: T) -> Tuple[T] =>
    @(0, p);

def main = {
    var a: Tuple[String] = make_zero[String]("Hello");

    print(a);

    print(a.0);
    print(a.1);
};
