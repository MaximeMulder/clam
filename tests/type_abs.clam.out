(0, "Hello")
0
"Hello"
